### PR TITLE
Multiple changes, mainly add login implementation

### PR DESCRIPTION
**Pull request by JohnStarich**
- add default content type to JsonApplication routes = 'application/json'
- add ObjectId serializer for Gson
- add parse method for JsonTransformer
- add default content type for html service wildcard route
- refactor email to username
- add login implementation

### CODE REVIEW

File: src/main/java/com/johnstarich/moviematcher/models/User.java [hunk 1]
```diff
@@ -94,63 +95,80 @@ public User resetPassword(String oldPassword, String newPassword) throws HttpExc
...
+		Optional<User> user = loadByUsername(username);
+		// username does not exist
+		if(! user.isPresent()) {
+			throw new HttpException(HttpStatus.UNAUTHORIZED, "Bad username/password combination. Please try again.");
+		}
+		// passwords do not match
+		if(! BCrypt.checkpw(password, user.get().password)) {
+			throw new HttpException(HttpStatus.UNAUTHORIZED, "Bad username/password combination. Please try again.");
+		}
+		return new Session(null, user.get()).save();
```
Comment (jjalfaro9) [inline src/main/java/com/johnstarich/moviematcher/models/User.java:123]: Why are you creating a session with a null objectId() ?

Comment (JohnStarich) [inline src/main/java/com/johnstarich/moviematcher/models/User.java:123]: It does the same thing as `new ObjectId()` but it's one less import
I can change it if you think I should, for consistency's sake or something

Comment (jjalfaro9) [inline src/main/java/com/johnstarich/moviematcher/models/User.java:123]: :+1:

**Comment by jjalfaro9** [comment]
LGTM :+1:

**Comment by jjalfaro9** [comment]
retry

**Pull request merged**